java/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -58,8 +59,24 @@ private static native Dataset writeWithFfiStream(long arrowStreamMemoryAddress,
   public native int countRows();
 
   /**
-   * Closes this dataset and releases any system resources associated with it.
-   * If the dataset is already closed, then invoking this method has no effect.
+   * Get all fragments in this dataset.
+   *
+   * @return A list of {@link Fragment}.
+   */
+  public List<Fragment> getFragments() {
+    // Set a pointer in Fragment to dataset, to make it is easier to issue IOs later.
+    //
+    // We do not need to close Fragments.
+    return this.getFragmentsNative().stream()
+        .peek(f -> f.dataset = this)
+        .collect(Collectors.toList());
+  }
```

Comment:
You could probably simplify this quite a bit if you wanted.  Since you only need the fragment ID from rust you could change `getFragmentsNative` to `getFragmentIdsNative` which just returns a stream of `long`.  Then you could construct the `Fragment` object and build up the `ArrayList` in this method (in a Java context).  Then you could pass in both the `Dataset` and the `long` as part of the `Fragment` constructor (so you could make the `dataset` field private).

Reply:
Good point. I changed `getFragmetns() -> int[]`.